src/components/UserCurrentLocationButton.js

```diff
@@ -0,0 +1,97 @@
+import PropTypes from 'prop-types';
+import React, {useEffect, useRef} from 'react';
+import {Text} from 'react-native';
+import getCurrentPosition from '../libs/getCurrentPosition';
+import * as User from '../libs/actions/User';
+import styles from '../styles/styles';
+import Icon from './Icon';
+import * as Expensicons from './Icon/Expensicons';
+import LocationErrorMessage from './LocationErrorMessage';
+import withLocalize, {withLocalizePropTypes} from './withLocalize';
+import colors from '../styles/colors';
+import PressableWithFeedback from './Pressable/PressableWithFeedback';
+
+const propTypes = {
+    /** Callback that runs when location data is fetched */
+    onLocationFetched: PropTypes.func.isRequired,
+
+    /** Boolean to indicate if the button is clickable */
+    isDisabled: PropTypes.bool,
+
+    ...withLocalizePropTypes,
+};
+
+const defaultProps = {
+    isDisabled: false,
+};
+
+function UserCurrentLocationButton({onLocationFetched, isDisabled, translate}) {
+    const isFetchingLocation = useRef(false);
+
+    /**
+     * handles error when failed to get user's current location
+     * @param {Object} errorData
+     * @param {Number} errorData.code
+     */
+    const onError = (errorData) => {
+        isFetchingLocation.current = false;
+
+        User.setLocationError(errorData.code);
+    };
+
+    /**
+     * handles success after getting user's current location
+     * @param {Object} successData
+     * @param {Object} successData.coords
+     * @param {Number} successData.coords.longitude
+     * @param {Number} successData.coords.latitude
+     * @param {Number} successData.timestamp
+     */
+    const onSuccess = (successData) => {
+        isFetchingLocation.current = false;
+
+        User.clearLocationError();
+
+        onLocationFetched(successData);
+    };
+
+    /** Gets the user's current location and registers success/error callbacks */
+    const useCurrentLocation = () => {
+        if (isFetchingLocation.current) return;
+
+        isFetchingLocation.current = true;
+
+        getCurrentPosition(onSuccess, onError, {
+            maximumAge: 0, // no cache, always get fresh location info
+            timeout: 3000,
```

Comment:
If `enableHighAccuracy` is causing issues for Android, we can disabled it (i.e. set it to false). Wifi/cell-tower signal is accurate enough for this feature.

timeout -> let's extend this to `5000`. We want to reduce the possibility for time out errors. `3000` might be little short.

maximumAge: 0 -> this is okay. Let's keep `0`

Reply:
@huzaifa-99 I'm always impressed the depth of your research 🙇 

I also see [react-native-geolocation-service](https://github.com/Agontuk/react-native-geolocation-service) is much more popular and widely used.

I don't know how prevent the timeout issue with Android though. Is the API same between `react-native-geolocation-service` and `@react-native-community/geolocation`? Is it possible we can just swap them?

If it's easy, let's change the library to `react-native-geolocation-service`

Otherwise, we can set `enableHighAccuracy` to `false` for now

Reply:
> I think we can set enableHighAccuracy to false for now (which is the default).

yep I agree 👍